microsoft/testsuites/core/floppy.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import RemoteNode, TestCaseMetadata, TestSuite, TestSuiteMetadata
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite ensures the floppy module is disabled.
+    """,
+)
+class Floppy(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Ensure the floppy module is not loaded.
+        """,
+        priority=0,
+    )
+    def check_floppy_module(self, node: RemoteNode) -> None:
+        cmd_result = node.execute("/usr/sbin/lsmod | /usr/bin/grep floppy", shell=True)
```

Comment:
use tool node.tools[Lsmod]

Reply:
here is behavior in centos79.
it is built as module.
cat /boot/config-3.10.0-1160.31.1.el7.x86_64 | grep -i CONFIG_BLK_DEV_FD
CONFIG_BLK_DEV_FD=m

We ask distro vendor to blacklist it.
cat /etc/modprobe.d/* | grep -i blacklist
blacklist amdgpu
**blacklist floppy**
blacklist nouveau
blacklist radeon
blacklist dccp
blacklist dccp_diag
blacklist dccp_ipv4
blacklist dccp_ipv6

Reply:
we should check it in Python, so avoid to use `|` in code, unless the output is over 1000 lines. It brings raw output to log, and is good for troubleshooting.

Reply:
I am using Lsmod tool now that bugs in the tool have been resolved.